src/Compilers/Core/Portable/InternalUtilities/ExceptionUtilities.cs

```diff
@@ -27,5 +28,14 @@ internal static Exception Unreachable
         {
             get { return new InvalidOperationException("This program location is thought to be unreachable."); }
         }
+
+        /// <summary>
+        /// Determine if an exception was an <see cref="OperationCanceledException"/>, and that the provided token caused the cancellation.
+        /// </summary>
+        /// <param name="exception">The exception to test.</param>
+        /// <param name="cancellationToken">Checked to see if the provided token was cancelled.</param>
+        /// <returns><see langword="true"/> if the exception was an <see cref="OperationCanceledException" /> and the token was canceled.</returns>
+        internal static bool IsCurrentOperationBeingCancelled(Exception exception, CancellationToken cancellationToken)
+            => exception is OperationCanceledException oce && cancellationToken.IsCancellationRequested;
```

Comment:
This could be true when `exception` and `cancellationToken` are unrelated instances. Is there a reason you aren't checking `&& oce.CancellationToken == cancellationToken` here? 

Effectively I would expect it to be written as such (do the prop check before type check as it's cheaper)

```suggestion
            => 
                cancellationToken.IsCancellationRequested && 
                exception is OperationCanceledException oce && 
                oce.CancellationToken == cancellationToken;
```

Reply:
➡️ The omission of cancellation identity check is intentional. See https://github.com/dotnet/roslyn/pull/58843#discussion_r789824160.

Reply:
```suggestion
            => exception is OperationCanceledException && cancellationToken.IsCancellationRequested;
```

I think I prefer this better as then there would be no allocation of a new object (as a copy of exception casted to that exception type) needlessly just for this check (increasing performance).